src/link.js

```diff
@@ -75,17 +61,69 @@ module.exports = function link(config, args) {
     asset => path.basename(asset)
   );
 
-  if (isEmpty(assets)) {
-    return;
-  }
+  const tasks = dependencies.map((dependency) => (next) => {
+    const before = (cb) => {
+      if (dependency.config.hooks && dependency.config.hooks.before) {
+        exec(dependency.config.hooks.before, function before(err, stdout, stderr) {
+          if (!err && !stderr) {
+            console.log(stdout);
+            cb();
+          } else {
+            throw Error(err || stderr);
+          }
+        });
+      } else {
+        cb();
+      }
+    };
 
-  if (project.ios) {
-    log.info('Linking assets to ios project');
-    copyAssetsIOS(assets, project.ios);
-  }
+    const after = (cb) => {
```

Comment:
Isn't it better to write:
```js
const after = done =>  (dependency.config.hooks || {}).before 
   ? exec(dependency.config.hooks.after, done)
    : done()
};
```
I would also suggest to add in `hooks` to a function that makes config in `rnpm` automatically so that we don't have to check if `dependency.config.hooks` is defined.